main_test.go

```diff
@@ -7,35 +7,74 @@ import (
 
 func TestParseTextToJSON(t *testing.T) {
 	tests := []struct {
-		name     string
-		input    string
-		expected map[string]interface{}
+		name           string
+		input          string
+		expectedOutput interface{}
+		expectedError  error
 	}{
 		{
 			name:  "Test simple parsing",
 			input: `name:"Fikar" age:10 address:{city:"Jakarta Selatan" state:"DKI Jakarta"} gender:"male" is_married:false`,
-			expected: map[string]interface{}{
-				"name": "Fikar",
-				"age":  10,
-				"address": map[string]interface{}{
-					"city":  "Jakarta Selatan",
-					"state": "DKI Jakarta",
-				},
+			expectedOutput: map[string]interface{}{
+				"name":       "Fikar",
+				"age":        10,
+				"address":    map[string]interface{}{"city": "Jakarta Selatan", "state": "DKI Jakarta"},
 				"gender":     "male",
 				"is_married": false,
 			},
+			expectedError: nil,
+		},
+		{
+			name:  "Test scientific notation number",
+			input: `name:"John" age:30 salary:2e+06 married:true`,
+			expectedOutput: map[string]interface{}{
+				"name":    "John",
+				"age":     30,
+				"salary":  2000000.0,
+				"married": true,
+			},
+			expectedError: nil,
+		},
+		{
+			name:  "Test identical field name, with simple value",
+			input: `name:"Alice" age:25 name:"Bob"`,
+			expectedOutput: map[string]interface{}{
+				"name": []interface{}{
+					"Alice",
+					"Bob",
+				},
+				"age": 25,
+			},
+			expectedError: nil,
+		},
+		{
+			name:  "Test identical field name, with object value",
+			input: `display_data:{header:"header_1" title:"title_1"} display_data:{header:"header_2"  title:"title_2"}`,
```

Comment:
LGTM. btw to check for _potential_ breaking when _child_ has the same key name as the _parent_, wdyt if we test this input?

```
display_data:{display_data:"display_data1"}
```

Reply:
test added.